generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/execution/FunctionDataFetcher.kt

```diff
@@ -49,7 +49,7 @@ open class FunctionDataFetcher(
     /**
      * Invoke a suspend function or blocking function, passing in the [target] if not null or default to using the source from the environment.
      */
-    override fun get(environment: DataFetchingEnvironment): Any? {
+    override fun get(environment: DataFetchingEnvironment): CompletableFuture<Any?> {
```

Comment:
i believe that this change is going to cause incompatiblity with the update to [graphql-java 22](https://github.com/graphql-java/graphql-java/releases/tag/v22.0)

Reply:
Can you help me understand more specifically what you mean so that maybe I can try to adjust my solution accordingly?

Otherwise, I'd say I'm happy to remove the change to this `get` function signature as it's not required for my solution to work. Would that be an okay compromise?

Reply:
TLDR; graphql-java 22 introduces polymorphic resolution, before even if your data fetcher was returning a materialized object (not a CF), it was anyways creating a CF creating a terrible memory bottleneck, and GC pause times were paying the price.

having said that, i wonder, why would you want to catch `Error`s ?
screenshot taken from `Error` docs
<img width="690" alt="image" src="https://github.com/ExpediaGroup/graphql-kotlin/assets/6611331/aec94427-6d52-4a78-8bc3-93c9e492ca92">

Reply:
> TLDR; graphql-java 22 introduces polymorphic resolution, in previous versions, even if your data fetcher was returning a materialized object (not a CF), it was anyways creating a CF causing a terrible memory bottleneck where GC pause times were paying the price.

😅  I need to read up on this, as tbh that is _pretty far out of my depth_ of current expertise lol. So I'm inclined to defer to your judgement and undo this part of my change regardless of where the larger change lands as accepted or not.

> having said that, i wonder, why would you want to catch Errors ?
screenshot taken from Error docs

In regards to this, please see my thoughts/discussion here https://github.com/ExpediaGroup/graphql-kotlin/pull/1966#issuecomment-2071060344 if you would.